### ISSUE TITLE
[Feature request] Regex in search

### ISSUE THREAD

**Issue by lexfrei** [amir20/dozzle#1103]
**Is your feature request related to a problem? Please describe.**
It's not possible to search containers with regex now. In large compose infra with container names like ProjectName_Branch_Service_Version, it's not easy to find exactly what you need

**Describe the solution you'd like**
Possibility to use regex in search

**Describe alternatives you've considered**
For now, I forked this repo and changed this to 100. After an initial search, ctrl+f and second search.
https://github.com/amir20/dozzle/blob/c8d5c9889214b2889d7c8ed39bd90ac4a1d290ab/assets/pages/Index.vue#L56

**Additional context**
Also, it would be great to have an option to set this amount manually.

**Comment by amir20** [comment]
@lexfrei interesting you bring this up. I was actually working on a better way to search all containers but abandoned the feature because I wasn't sure what it should look like. 

I have a few of questions for you: 

1. Are you ONLY talking about the index page? The code you referred is only the home page. 
2. How many containers do you have?
3. What kind of regex did you have in mind? Can you give me an idea as I am trying to find a good use case. 
4. Can you show me some real examples of your container names? 

I want to implement something with [fuse.js](https://fusejs.io/) or [fuzzysort](https://www.npmjs.com/package/fuzzysort). In addition to fixing it on the home page, I also plan to provide a shortcut like `command/ctrl+k` to bring up a simple search dialog.

**Comment by lexfrei** [comment]
1. Yep
2. ~50 per project. 3+ projects at once and some technical (like monitoring)
4. Something like `3.2*traefik` for next example.
3. `dummyProject_3.2_serviceName_subserviceName`

**Comment by amir20** [comment]
In about 15 minutes latest-dev should be finished building. I implemented a fuzzy search with fuzzysort. I tried it out and it works well. 

With `dummyProject_3.2_serviceName_traefik` I was able to filter the right item with `32st` which is pretty great. 

Give it a try. I am working on a better search integration.

**Comment by lexfrei** [comment]
Looks good!
Also, are there any options to see more than 10 containers at once?

**Comment by amir20** [comment]
I am still thinking about that. I show all containers (including stopped) which means it could be over 1000s. I am thinking on homepage to show 10 with a scroll and `command+k` would be a popup with more space to show more.

**Comment by amir20** [comment]
I just released a new official build `v3.4`. It has a search option now which works much better. 

I am going to come back to the limit by porting the new search functionality to the home page. 

@lexfrei I am going to close this for now. Thanks again for this.

**Comment by lexfrei** [comment]
I'm here to give you some feedback:
1. For now we have 3 different search boxes and it's really confusing. My suggestion: Merge them all in one search box with selectable modifications, like in VSCode
![image](https://user-images.githubusercontent.com/1366514/112800667-9f6b0b80-9078-11eb-8b35-dc44dc29d528.png)
2. ctrl+f search box doesn't work at all for me. The box appears and I can fill it, but nothing happens. Also, the `cmd+k` hint isn't useful on non-mac systems, so it would be great to add `ctrl+f` notation (btw, this doesn't work for me if the search pop-up is active)
3. I really need an option to see more containers at once on the main page. This feels native but missed. Probably, it can be done by dropdown menu with selectable limits if you think this will be slow or havy.

**Comment by amir20** [comment]
Hi @lexfrei. Thanks for the feedback. I am still not understanding some of your questions though. See if you can help me. 

> we have 3 different search boxes and it's really confusing ... Merge them all in one search box with selectable modifications, like in VSCode

I don't understand your point here. There should only be two kind of search boxes. There is the search with `ctrl+f` which searches through logs. Then there is fuzzy container name search which is `ctrl+k`. Where is the third one? VSCode does exactly the same thing. It has "Go to file" with `shift+meta+O`. I am not counting the homepage search because that's just a landing page and I plan to replace it with the fuzzy search. 


> ctrl+f search box doesn't work at all for me

This works for me. I just tested it. 

<img width="1225" alt="Screen Shot 2021-03-29 at 1 21 57 PM" src="https://user-images.githubusercontent.com/260667/112895311-cf72d880-9091-11eb-8df8-a3730dcfb391.png">

Are you getting javascript console errors? Please define what is not working. It might be a bug and you should create a new bug if it is a browser I don't test usually. 

> Also, the cmd+k hint isn't useful on non-mac systems, so it would be great to add ctrl+f notation 

I don't know what you mean. It is `cmd+k` or `ctrl+k`. There is also a new search icon. 

<img width="402" alt="Screen Shot 2021-03-29 at 1 23 55 PM" src="https://user-images.githubusercontent.com/260667/112895601-30021580-9092-11eb-8155-1f0964bfb5c8.png">

The shortcut is also on the tooltip and input placeholder. 

<img width="626" alt="Screen Shot 2021-03-29 at 1 25 30 PM" src="https://user-images.githubusercontent.com/260667/112895670-41e3b880-9092-11eb-8974-8a9664845f88.png">

> btw, this doesn't work for me if the search pop-up is active

That's to be expected. Why would you expect it to work? You have to be outside of the input box to capture any shortcuts. 

> I really need an option to see more containers at once on the main page. This feels native but missed. Probably, it can be done by dropdown menu with selectable limits if you think this will be slow or heavy.

I still don't understand why you need this ON the homepage. The new search functionality should be able to search 1000s of container names and filter to top choices. As I said, I am working on replacing the homepage with container fuzzy search component. But why do you need more containers? If your query is well defined then shouldn't it be on top 10? 

I think I am missing something important on the last one. You shouldn't need to go back to home page to search every time. That's the whole reason why I added the search functionality.  Based on your number 2 comment, I am guessing you haven't used it.


In summary...

1. There should only be two kinds of searches. One that search logs and another that searches containers. I am trying to mirror VSCode here. 
2. Search should work. If it doesn't, open a bug. You should also be able to trigger fuzzy search with `ctrl+k` or clicking the search icon. 
3. I don't know why you need the homepage to show everything. I imagine people jumping between containers and never visiting homepage again.

**Comment by lexfrei** [comment]
Yep! What's what I'm talking about :) Too many different searches and it's confusing :)
![image](https://user-images.githubusercontent.com/1366514/112957871-8b8fda00-914a-11eb-9a3b-9847c54e8ef6.png)
I didn't get your logic but now it's clear to me

**Comment by amir20** [comment]
That's probably a bug where you see search 3 on homepage. I removed it to only container pages with 0a0f6b5.
